src/town_cmd.cpp

```diff
@@ -1087,6 +1087,49 @@ static bool GrowTownWithRoad(const Town *t, TileIndex tile, RoadBits rcmd)
 	return false;
 }
 
+/**
+ * Checks if a town road can be continued on the other side of a bridge.
+ *
+ * @param end_tile The end tile of the bridge
+ * @param bridge_dir The direction of the bridge
+ * @return true if the road can be continued, else false
+ */
+static bool CanRoadContinueAfterBridge(const Town* t, const TileIndex end_tile, const DiagDirection bridge_dir)
+{
+	const int delta = TileOffsByDiagDir(bridge_dir); // +1 tile in the direction of the bridge
+	TileIndex next_tile = end_tile + delta; // The tile beyond the bridge
+	RoadBits rcmd = DiagDirToRoadBits(ReverseDiagDir(bridge_dir)); // The road bits that would be built after extending the bridge.
+	RoadType rt = GetTownRoadType(t); // The road type being built by the town.
+
+	/* Before we try anything, make sure the tile is on the map and not the void. */
+	if (!IsValidTile(next_tile)) return false;
+
+	/* If the next tile is a bridge or tunnel, allow if it's a road bridge/tunnel continuing in the same direction. */
+	if (IsTileType(next_tile, MP_TUNNELBRIDGE)) {
+		return GetTunnelBridgeTransportType(next_tile) == TRANSPORT_ROAD && GetTunnelBridgeDirection(next_tile) == bridge_dir;
+	}
+
+	/* If the next tile is a station, allow if it's a road station facing the proper direction. Otherwise return false. */
+	if (IsTileType(next_tile, MP_STATION)) {
+		/* If the next tile is a road station, allow if it's facing the same direction, otherwise disallow. */
+		return IsRoadStop(next_tile) && GetRoadStopDir(next_tile) == ReverseDiagDir(bridge_dir);
+	}
+
+	/* If the next tile is a railroad track, check if towns are allowed to build level crossings. If both are true, reject. Else allow DoCommand to determine if the rail track is buildable. */
+	if (IsTileType(next_tile, MP_RAILWAY) && !_settings_game.economy.allow_town_level_crossings) return false;
+
+	/* If a road tile can be built, the bridge is allowed. */
+	if (DoCommand(next_tile, rcmd | (rt << 4), t->index, DC_AUTO | DC_NO_WATER, CMD_BUILD_ROAD).Succeeded()) return true;
+
+	/* If the next tile is a road depot, allow if it's facing the new bridge. */
+	if (IsTileType(next_tile, MP_ROAD)) {
+		return IsRoadDepot(next_tile) && GetRoadDepotDirection(next_tile) == ReverseDiagDir(bridge_dir);
+	}
```

Comment:
If possible, maybe better to put this one right below `MP_STATION` ? That means you can just do: `return DoCommand`, making the function even more readable :) But I do not know if this is possible :)

Reply:
Yes, good suggestion.